File: jsweet-client/src/main/java/json_schema/control/StartGame.java
package json_schema.control;

import json_schema.GameSetup2;
import json_schema.comm.BotAddress;
import def.js.Array;

public class StartGame extends Command {

	public static final String TYPE = "startGame";

	public StartGame() {
		super(TYPE);
	}

	public StartGame(String type) {
		super(type);
	}

	public void setGameSetup(GameSetup2 gameSetup) {
		$set("gameSetup", gameSetup);
	}

	public void setBotAddresses(Array<BotAddress> botAddresses) {
		$set("botAddresses", botAddresses);
	}
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/control/StartGame.kt
package net.robocode2.gui.model.control

import net.robocode2.gui.model.ClientContent
import net.robocode2.gui.model.ContentType
import net.robocode2.gui.model.GameSetup
import net.robocode2.gui.model.comm.BotAddress

class StartGame(
        clientKey: String,
        val gameSetup: GameSetup,
        val botAddresses: Set<BotAddress>
) : ClientContent(type = ContentType.START_GAME.type, clientKey = clientKey)